keyboards/uno/keymaps/default/keymap.c

```diff
@@ -0,0 +1,102 @@
+/* Copyright 2020 Snipeye
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+enum uno_keycode
+{
+  UNO = SAFE_RANGE
+};
+
+static uint16_t pressTimer = 0xFFFF;
+#define CUSTOM_LONGPRESS 150
+#define CUSTOM_LONGERPRESS 750
+#define CUSTOM_STRING "I can put a whole buncha text in here and type it all with a single keypress."
+#define RESET_LENGTH 3000
+const uint8_t PROGMEM RGBLED_RAINBOW_MOOD_INTERVALS[] = { 10, 25, 50 };
+
+char stringToSend[2] = "a";
+char maxLetter = 'z';
+
+uint8_t presetCounter = 0;
+
+#define COUNTER X_A
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+      [0] = LAYOUT(
+            UNO
+          )
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
```

Comment:
It looks like most of this could be handled by the tap dance feature. 
https://docs.qmk.fm/#/feature_tap_dance

Reply:
I don't think the tap dance feature is really appropriate here. It's not really ideal for situations where you have more than 2 or 3 features. There's also some unique behavior here that can't be emulated by tap dance.

Reply:
The advanced tap dances support functions, so yes, it does support sending strings. Eg: 
https://docs.qmk.fm/#/feature_tap_dance?id=example-4


As for the unique behavior?  do you mean the custom tapping term? If so, then, well, tap dance also supports that with the per key tapping term. 

It's just that this looks a LOT like what tap dance is capable of, and I'd prefer that people didn't "reinvent" the wheel when there is a perfectly good solution already available. 
But if you're happy with your code, then that's fine by me.